src/Object/Klip/KlipSchema.php

```diff
@@ -11,55 +11,133 @@ class KlipSchema extends BaseApiResource
     const FIELD_TITLE = 'title';
     const FIELD_WORKSPACE = 'workspace';
     const FIELD_COMPONENTS = 'components';
+    const FIELD_DATASOURCES = 'datasources';
+
+
+    /**
+     * BaseApiResource constructor.
+     * @param array $data
+     */
+    public function __construct(array $data = [])
+    {
+        parent::__construct($data);
+
+        if (!isset($this->data['schema'][static::FIELD_WORKSPACE])) {
+            $this->data['schema'][static::FIELD_WORKSPACE] = [];
+        }
+
+        // Add empty datasource array to object if it doesn't exist yet
+        if (!array_key_exists(static::FIELD_DATASOURCES, $this->getWorkspace()) || !is_array($this->getWorkspace()[static::FIELD_DATASOURCES])) {
+            $workspace = $this->getWorkspace();
+            $workspace[static::FIELD_DATASOURCES] = [];
+            $this->setWorkspace($workspace);
+        }
+
+    }
 
     /**
      * @return mixed
      */
-    public function getTitle() {
-        return $this->data[self::FIELD_TITLE];
+    public function getTitle()
+    {
+        return $this->data['schema'][static::FIELD_TITLE];
```

Comment:
According to this: http://apidocs.klipfolio.com/docs/klips-schema Only get GET request returns the data inside a schema key.
Perhaps we could check in the constructor if it's inside a schema key and move the data out of it so that we're able to PUT the same data.